cmd/main_test.go

```diff
@@ -399,3 +401,44 @@ func TestRun_RerunFails_PanicPreventsRerun(t *testing.T) {
 	err := run(opts)
 	assert.ErrorContains(t, err, "rerun aborted because previous run had a suspected panic", out.String())
 }
+
+func TestExecStdin(t *testing.T) {
+	stdin := os.Stdin
+	t.Cleanup(func() { os.Stdin = stdin })
+
+	r, w, err := os.Pipe()
+	assert.NilError(t, err)
+	defer r.Close()
+
+	os.Stdin = r
+
+	go func() {
+		defer w.Close()
+		e := json.NewEncoder(w)
+		for _, event := range []struct {
+			Time    *time.Time `json:",omitempty"`
+			Action  string
+			Package string  `json:",omitempty"`
+			Test    string  `json:",omitempty"`
+			Elapsed float64 `json:",omitempty"`
+			Output  string  `json:",omitempty"`
+		}{
+			{Action: "run", Package: "pkg"},
+			{Action: "run", Package: "pkg", Test: "TestOne"},
+			{Action: "fail", Package: "pkg", Test: "TestOne"},
+			{Action: "fail", Package: "pkg"},
+		} {
+			assert.NilError(t, e.Encode(event))
```

Comment:
I think `t.Fatal` (and anything that results in `t.FailNow`) is only supposed to be called from the main test goroutine. Here `assert.NilError` will call `t.FailNow`.  One way to work around this is to use `t.Error` (`t.Fail`) instead, which can be done with `assert.Check(t, e.Encode(event))`.